tempodb/wal/wal.go

```diff
@@ -90,7 +89,12 @@ func (w *WAL) RescanBlocks(log log.Logger) ([]*AppendBlock, error) {
 		}
 
 		start := time.Now()
-		level.Info(log).Log("msg", "beginning replay", "file", f.Name(), "size", f.Size())
+		fileInfo, err := f.Info()
+		if err != nil {
+			return nil, err
+		}
+
+		level.Info(log).Log("msg", "beginning replay", "file", f.Name(), "size", fileInfo.Size())
```

Comment:
@kvrhdn Done 😃, please take a look again. The `err` here is not `nil` when the file `f` is remove after `os.ReadDir` is called, it's a rare error but still possible sometimes.